Batteries/Control/Lawful/MonadLift.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2025 Quang Dao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Quang Dao
+-/
+
+/-!
+# Lawful version of `MonadLift`
+
+This file defines the `LawfulMonadLift(T)` class, which adds laws to the `MonadLift(T)` class.
+-/
+
+universe u v w
+
+/-- The `MonadLift` typeclass only contains the lifting operation. `LawfulMonadLift` further
+  asserts that lifting commutes with `pure` and `bind`:
+```
+monadLift (pure a) = pure a
+monadLift ma >>= monadLift ∘ f = monadLift (ma >>= f)
+```
+-/
+class LawfulMonadLift (m : semiOutParam (Type u → Type v)) (n : Type u → Type w)
+    [Monad m] [Monad n] [inst : MonadLift m n] : Prop where
+  /-- Lifting preserves `pure` -/
+  monadLift_pure {α : Type u} (a : α) : inst.monadLift (pure a) = pure a
```

Comment:
Not necessary, but another way to write this is:
```
  monadLift_pure {α : Type u} (a : α) : monadLift (pure a : m α) = (pure a : n α)
```
Perhaps it is clearer? Not sure.